contracts/TransferLimitModule.sol

```diff
@@ -0,0 +1,417 @@
+        require(_dutchxAddr != 0, "Invalid dutchx address");
+
+        timePeriod = _timePeriod;
+        rolling = _rolling;
+        globalWeiCap = _globalWeiCap;
+        globalDaiCap = _globalDaiCap;
+        threshold = _threshold;
+        delegate = _delegate;
+        dutchx = DutchExchange(_dutchxAddr);
+
+        for (uint256 i = 0; i < tokens.length; i++) {
+            transferLimits[tokens[i]].transferLimit = _transferLimits[i];
+        }
+    }
+
+    /// @dev Allows to update the transfer limit for a specified token. This can only be done via a Safe transaction.
+    /// @param token Token contract address.
+    /// @param transferLimit Transfer limit in smallest token unit.
+    function changeTransferLimit(address token, uint256 transferLimit)
+        public
+        authorized
+    {
+        transferLimits[token].transferLimit = transferLimit;
+    }
+
+    /// @dev Updates the delegate. This can only be done via a Safe transaction.
+    /// @param _delegate New delegate, who is a non-owner account also allowed to make transfers.
+    function setDelegate(address _delegate)
+        public
+        authorized
+    {
+        delegate = _delegate;
+    }
+
+    /// @dev Returns if Safe transaction is a valid transfer limit transaction.
+    /// @param token Address of the token that should be transfered (0 for Ether)
+    /// @param to Address to which the tokens should be transfered
+    /// @param amount Amount of tokens (or Wei) that should be transfered
+    /// @param safeTxGas Gas that should be used for the Safe transaction.
+    /// @param dataGas Gas costs for data used to trigger the safe transaction and to pay the payment transfer
+    /// @param gasPrice Gas price that should be used for the payment calculation.
+    /// @param gasToken Token address (or 0 if ETH) that is used for the payment.
+    /// @param refundReceiver Address of receiver of gas payment (or 0 if tx.origin).
+    /// @param signatures Packed signature data ({bytes32 r}{bytes32 s}{uint8 v})
+    /// @return Returns if transaction can be executed.
+    function executeTransferLimit(
+        address token,
+        address to,
+        uint256 amount,
+        uint256 safeTxGas,
+        uint256 dataGas,
+        uint256 gasPrice,
+        address gasToken,
+        address refundReceiver,
+        bytes signatures
+    )
+        public
+    {
+        require(to != 0, "Invalid to address provided");
+        require(amount > 0, "Invalid amount provided");
+
+        uint256 startGas = gasleft();
+        bytes32 txHash = getTransactionHash(
+            token, to, amount,
+            safeTxGas, dataGas, gasPrice, gasToken, refundReceiver,
+            nonce
+        );
+        require(checkSignatures(txHash, signatures), "Invalid signatures provided");
+        // Increase nonce and execute transaction.
+        nonce++;
+        require(gasleft() >= safeTxGas, "Not enough gas to execute safe transaction");
```

Comment:
This check should probably be done after we call `handleTransferLimits` (since that also uses up some gas)